server/src/main/scala/io/delta/standalone/internal/DeltaSharedTableLoader.scala

```diff
@@ -180,18 +182,87 @@ class DeltaSharedTable(
     snapshot.version -> actions
   }
 
-  def queryCDF(
-    cdfOptions: Map[String, String]
-  ): (Long, Seq[model.SingleAction]) = withClassLoader {
-    throw new IllegalStateException("queryCDF is not supported yet")
+
+  def queryCDF(cdfOptions: Map[String, String]): Seq[model.SingleAction] = withClassLoader {
+    val actions = ListBuffer[model.SingleAction]()
+
+    // First: get Protocol and Metadata
+    val snapshot = deltaLog.snapshot
+    val modelProtocol = model.Protocol(snapshot.protocolScala.minReaderVersion)
+    val modelMetadata = model.Metadata(
+      id = snapshot.metadataScala.id,
+      name = snapshot.metadataScala.name,
+      description = snapshot.metadataScala.description,
+      format = model.Format(),
+      schemaString = cleanUpTableSchema(snapshot.metadataScala.schemaString),
+      configuration = getMetadataConfiguration(snapshot.metadataScala.configuration),
+      partitionColumns = snapshot.metadataScala.partitionColumns
+    )
+    actions.append(modelProtocol.wrap)
+    actions.append(modelMetadata.wrap)
+
+    // Second: get files
+    val cdcReader = new DeltaSharingCDCReader(deltaLog, conf)
+    val (changeFiles, addFiles, removeFiles) = cdcReader.queryCDF(cdfOptions, tableVersion)
+    changeFiles.foreach{cdcDataSpec =>
```

Comment:
do we need spaces around brackets... perhaps run lint?

Reply:
updated. lint rans on testes, actually it didn't complain.